src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Services/DatalockValidationService.cs

```diff
@@ -0,0 +1,295 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using SFA.DAS.CollectionEarnings.DataLock.Application.DataLock;
+using SFA.DAS.CollectionEarnings.DataLock.Application.DataLock.Matcher;
+using SFA.DAS.CollectionEarnings.DataLock.Domain;
+using SFA.DAS.CollectionEarnings.DataLock.Domain.Extensions;
+using SFA.DAS.CollectionEarnings.DataLock.Infrastructure.Data.Entities;
+using SFA.DAS.Payments.DCFS.Domain;
+using SFA.DAS.ProviderPayments.Calc.Shared.Infrastructure.Data.Entities;
+
+namespace SFA.DAS.CollectionEarnings.DataLock.Services
+{
+    public interface IValidateDatalocks
+    {
+        DatalockValidationResult ValidateDatalockForProvider(
+            ProviderCommitments providerCommitments, 
+            IEnumerable<RawEarning> providerEarnings, 
+            ImmutableHashSet<long> accountsWithNonPayableFlagSet);
+    }
+
+    public class DatalockValidationService : IValidateDatalocks
```

Comment:
This class is processing by Provider, but there are where clauses inside the code which are filtering by UKPRN, shouldn't the data being passed into this service already do this? It may simplify the logic